app/models/katello/content_view_component.rb

```diff
@@ -2,23 +2,80 @@ module Katello
   class ContentViewComponent < Katello::Model
     self.include_root_in_json = false
 
-    belongs_to :content_view, :class_name => "Katello::ContentView",
+    belongs_to :composite_content_view, :class_name => "Katello::ContentView",
                               :inverse_of => :content_view_components
     belongs_to :content_view_version, :class_name => "Katello::ContentViewVersion",
                                       :inverse_of => :content_view_components
+    belongs_to :content_view, :class_name => "Katello::ContentView"
 
     validates_lengths_from_database
-    validates :content_view_version_id, :uniqueness => {:scope => :content_view_id}
+    validates :content_view_version_id, :uniqueness => {:scope => :composite_content_view_id}, :if => proc { |r| r.content_view_version_id.present? }
+    validates :content_view_id, :uniqueness => {:scope => :composite_content_view_id}, :if => proc { |r| r.content_view_id.present? }
+    validates :composite_content_view, :presence => true
+
     validate :content_view_types
 
+    before_validation :update_content_view
+
+    def latest_version
+      if latest?
+        self.content_view.versions.last
+      else
+        self.content_view_version
+      end
+    end
+
     private
 
     def content_view_types
```

Comment:
Also, mind renaming this to something like `ensure_valid_attributes` or some such?